src/FakePdo.php

```diff
@@ -97,20 +100,62 @@ public function lastInsertId($seqname = null) : string
         return $this->lastInsertId;
     }
 
+	/**
+	 * @return bool
+	 */
     public function beginTransaction()
     {
         Server::snapshot('transaction');
         return true;
     }
 
+	/**
+	 * @return bool
+	 */
     public function commit()
     {
         return Server::deleteSnapshot('transaction');
     }
 
+	/**
+	 * @return bool
+	 * @throws Processor\ProcessorException
+	 */
     public function rollback()
     {
         Server::restoreSnapshot('transaction');
         return true;
     }
+
+	/**
+	 * @param string $statement
+	 * @return int|false
+	 */
+	public function exec($statement)
+	{
+		$statement = trim($statement);
+		if (strpos($statement, 'SET ')===0){
+			return false;
+		}
+
+		$sth = $this->prepare($statement);
+		if ($sth->execute()){
+			return $sth->rowCount();
+		}
+		return false;
+	}
+
+	/**
+	 * @param string $statement
+	 * @param int $mode
+	 * @param null $arg3
+	 * @param array $ctorargs
+	 * @return Php7\FakePdoStatement|Php8\FakePdoStatement
+	 */
+	public function query($statement, $mode = PDO::ATTR_DEFAULT_FETCH_MODE, $arg3 = null, array $ctorargs = [])
+	{
+		$sth = $this->prepare($statement);
+		$sth->execute();
+		return $sth;
```

Comment:
Not certain if anything new is needed here to replicate internal logic but this does run properly